scripts/helpers/function-helpers.ts

```diff
@@ -0,0 +1,135 @@
+import { Keyring } from "@polkadot/api";
+import { blake2AsHex } from "@polkadot/util-crypto";
+import { hexToU8a } from "@polkadot/util";
+
+export function schedulerWrapper(api, atBlock, tx) {
+  return api.tx.scheduler.schedule(atBlock, null, 0, tx);
+}
+
+export async function accountWrapper(api, privateKey) {
+  // Keyring
+  const keyring = new Keyring({ type: "ethereum" });
+
+  // Create account and get nonce
+  let account = await keyring.addFromUri(privateKey, null, "ethereum");
+  const { nonce: rawNonce } = (await api.query.system.account(account.address)) as any;
+  let nonce = BigInt(rawNonce.toString());
+
+  return [account, nonce];
+}
+
+export async function sudoWrapper(api, tx, account) {
+  try {
+    await api.tx.sudo.sudo(tx).signAndSend(account);
+    console.log("--> Sudo Tx sent\n");
+  } catch (e) {
+    console.log(e.message);
+  }
+}
+
+export async function preimageWrapper(api, tx, account, nonce) {
+  // Prepare the proposal hash
+  let preimage = {};
+  preimage["encodedProposal"] = tx?.method.toHex() || "";
+  preimage["encodedLength"] = hexToU8a(preimage["encodedProposal"]).length;
+  preimage["encodedHash"] = blake2AsHex(preimage["encodedProposal"]);
+  console.log("Encoded proposal hash is %s", preimage["encodedHash"]);
+  console.log("Encoded length %d", preimage["encodedLength"]);
+
+  // Preimage depending on RT version
+  let runtimeVersion = BigInt(api.runtimeVersion.specVersion);
+
+  // Check if preimage exists
+  let preimageCheck;
+  if (runtimeVersion < 2000n) {
+    preimageCheck = await api.query.democracy.preimages(preimage["encodedHash"]);
+  } else {
+    preimageCheck = await api.query.preimage.preimageFor([
+      preimage["encodedHash"],
+      preimage["encodedLength"],
+    ]);
+  }
+
+  // If preimage does not exists
+  if (preimageCheck.toHex() == "0x") {
+    try {
+      if (runtimeVersion < 2000n) {
+        preimageCheck = await api.query.democracy.preimages(preimage["encodedHash"]);
```

Comment:
I think you can remove this one, as you are fetching it above

Reply:
Yeah but here is to check whether the calculated preimage hash exists. If it does exist we don't have to send the preimage tx (we did that before and it would just fail). If it does not find it, it means that it does not exists, so we can send the tx.

Is a minor optimization but I guess it saves 1 tx from being sent if the preimage already exists. Not sure if it would be helpful in production, but why not hehe

Reply:
Sounds good! we can think about more optimization improvements after this pr too